R/ReportDocument.R

```diff
@@ -58,6 +58,56 @@ c.ReportDocument <- function(...) {
   out
 }
 
+#' @export
+metadata <- function(object, which = NULL) {
+  checkmate::assert_string(which, null.ok = TRUE)
+  UseMethod("metadata", object)
+}
+
+#' @export
+metadata.ReportDocument <- function(object, which = NULL) {
+  metadata <- attr(object, which = "metadata", exact = TRUE)
+  result <- metadata %||% list()
+  if (is.null(which)) {
+    return(result)
+  }
+  result[[which]]
+}
+
+#' @export
+metadata.ReportCard <- function(object, which = NULL) {
+  # TODO: soft deprecate
+  result <- list(title = object$get_name(), id = object$get_id())
```

Comment:
I think `id` not needed anymore? id is now a name of the object in the Report$cards list.

```suggestion
  result <- list(title = object$get_name())
```